### PR TITLE
feat(lint): Refine the eslint config + rules

### PR DESCRIPTION
**Pull request by seve**
* adds
  *  airbnb eslint plugin
  *  Disallow FC use
* modifies
  *  Moves prettier plugin to end of configs
* removes
  * Unneeded rule disabling

### CODE REVIEW

**Review by ovalenzuela19 — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -51,6 +39,18 @@ module.exports = {
         natural: false,
       },
     ],
+    "@typescript-eslint/ban-types": [
+      "error",
+      {
+        types: {
+          "React.FC": {
```
Comment (tihuan) [inline .eslintrc.js:46]: I did `yarn install` and tried to trigger this error by doing:

```tsx
export const Button: React.FC = () => {
  return <button>hi</button>;
};
```

but ESlint didn't error out!

If you've verified that it works, then I think we're good to merge and I'll figure out my IDE later lol

**Review by tihuan — COMMENTED** [review]
When I ran `yarn && yarn lint` I get the following error:

Screen Shot 2021-03-09 at 10 08 07 AM

Did you encounter that? Thanks!

File: tsconfig.json [hunk 2]
```diff
@@ -49,8 +49,5 @@
     // "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
     // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
   },
-  "include": [
-    "src/**/*",
-    "src/core/Button/.stories.tsx"
-  ]
+  "include": ["src/**/*", ".eslintrc.js"]
```
Comment (seve) [inline tsconfig.json:52]: Just want to make sure that this change is ok @tihuan

**Review by tihuan — APPROVED** [review]
Ahh got it! Great catch 👍 💯 Thanks so much, Seve! LGTMMM

**Pull request merged**